_inc/client/components/data/query-site/index.jsx

```diff
@@ -1,40 +1,55 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
 
 /**
  * Internal dependencies
  */
 import {
 	fetchSiteData,
-	isFetchingSiteData
+	isFetchingSiteData,
+	getSitePlan
 } from 'state/site';
 import { isDevMode } from 'state/connection';
 
-export const QuerySite = React.createClass( {
-	componentDidMount() {
-		if ( ! ( this.props.isFetchingSiteData || this.props.isDevMode ) ) {
+class QuerySite extends Component {
+	static propTypes = {
+		isFetchingSiteData: PropTypes.bool,
+		isDevMode: PropTypes.bool,
+		sitePlan: PropTypes.object
+	};
+
+	static defaultProps = {
+		isFetchingSiteData: false,
+		isDevMode: false,
+		sitePlan: {}
```

Comment:
This means we need to pass the sitePlan through evert `<QuerySite />` component we use, else it will always be empty and always refetch.  

I'm seeing the data fetch on tab change still because of this.